addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java

```diff
@@ -131,18 +133,46 @@ public void sessionChanged() {
     }
 
     public void optionsLoaded() {
+        currentServerUrl = param.getServerUrl();
         currentPollingFrequency = param.getPollingFrequency();
     }
 
     @Override
     public void optionsChanged(OptionsParam optionsParam) {
-        if (currentPollingFrequency != param.getPollingFrequency()) {
-            stopPoller();
-            startService();
-            currentPollingFrequency = param.getPollingFrequency();
-            LOGGER.debug(
-                    "Updated Interactsh Polling frequency to {} seconds.", currentPollingFrequency);
+        if (!StringUtils.equals(currentServerUrl, param.getServerUrl())) {
+            try {
+                stopPoller();
+                deregister();
```

Comment:
`stopPoller` could be moved to `deregister`, less error prone (that's duplicated in other place).